packages/access-api/src/service/upload-api-proxy.js

```diff
@@ -0,0 +1,266 @@
+import * as Client from '@ucanto/client'
+import * as CAR from '@ucanto/transport/car'
+import * as CBOR from '@ucanto/transport/cbor'
+// eslint-disable-next-line no-unused-vars
+import * as dagUcan from '@ipld/dag-ucan'
+import * as dagUcanDid from '@ipld/dag-ucan/did'
+import * as HTTP from '@ucanto/transport/http'
+// eslint-disable-next-line no-unused-vars
+import * as Store from '@web3-storage/capabilities/store'
+// eslint-disable-next-line no-unused-vars
+import * as Upload from '@web3-storage/capabilities/upload'
+// eslint-disable-next-line no-unused-vars
+import * as iucanto from '@ucanto/interface'
+// eslint-disable-next-line no-unused-vars
+import * as ed25519 from '@ucanto/principal/ed25519'
+
+/**
+ * @template {iucanto.Capability} C
+ * @template [Success=unknown]
+ * @template {{ error: true }} [Failure={error:true}]
+ * @callback InvocationResponder
+ * @param {iucanto.Invocation<C>} invocationIn
+ * @param {iucanto.InvocationContext} context
+ * @returns {Promise<iucanto.Result<Success, Failure>>}
+ */
+
+/**
+ * @typedef StoreService
+ * @property {InvocationResponder<iucanto.InferInvokedCapability<typeof Store.add>>} add
+ * @property {InvocationResponder<iucanto.InferInvokedCapability<typeof Store.list>>} list
+ * @property {InvocationResponder<iucanto.InferInvokedCapability<typeof Store.remove>>} remove
+ */
+
+/**
+ * @typedef UploadService
+ * @property {InvocationResponder<iucanto.InferInvokedCapability<typeof Upload.add>>} add
+ * @property {InvocationResponder<iucanto.InferInvokedCapability<typeof Upload.list>>} list
+ * @property {InvocationResponder<iucanto.InferInvokedCapability<typeof Upload.remove>>} remove
+ */
+
+/**
+ * @template {iucanto.Capability} C
+ * @template [Success=unknown]
+ * @template {{ error: true }} [Failure={error:true}]
+ * @param {object} options
+ * @param {Pick<Map<dagUcan.DID, iucanto.ConnectionView<any>>, 'get'>} options.connections
+ * @param {iucanto.Signer} [options.signer]
+ * @returns {InvocationResponder<C, Success, Failure>}
+ */
+function createInvocationResponder(options) {
+  /**
+   * @template {import('@ucanto/interface').Capability} Capability
+   * @param {iucanto.Invocation<Capability>} invocationIn
+   * @param {iucanto.InvocationContext} context
+   * @returns {Promise<iucanto.Result<any, { error: true }>>}
```

Comment:
great find. I agree on avoiding `any`.

I can't remove this `@returns` while also appeasing the `@returns {InvocationResponder<C, Success, Failure>}` on `createInvocationResponder` itself. Because that's generic, it's probalby fine/useful to have a generic `@returns` right?

I will update it so this inner `@returns` is `Promise<Ucanto.Result<Success, Failure>>`, which works and is much more generic.

if you prefer to remove the (now generic) `@returns` alltogether, say so and I'll do it here, or feel free to just remove it as fast follow.

Reply:
https://github.com/web3-storage/w3protocol/pull/334/commits/b9a3ed8496639920e98a3f0da3f2fbba0c8c7404#diff-969dad085be9e4a6140a50e5f0142a0e6f106b12a5c3375f5857006fbe71dab8R53